src/book/chapter_1_examples.clj

```diff
@@ -1,11 +1,178 @@
+;;and print a message like, “All tests passed.”
+;;
+;;Let’s see what it does. Pregnancy data from Cycle 6 of the NSFG is in a file called `2002FemPreg.dat.gz`; it is a gzip-compressed data file in plain text (ASCII), with fixed width columns.
+;;Each line in the file is a **record** that contains data about one pregnancy.
+;;
+;;The format of the file is documented in `2002FemPreg.dct`, which is a Stata dictionary file. Stata is a statistical software system; a “dictionary” in this context is a list of variable names,
+;;types, and indices that identify where in each line to find each variable.
+;;
+;;For example, here are a few lines from `2002FemPreg.dct`:
+;;
+;;```
+;;infile dictionary {
+;;   _column(1)  str12  caseid    %12s  "RESPONDENT ID NUMBER"
+;;   _column(13) byte   pregordr   %2f  "PREGNANCY ORDER (NUMBER)"
+;; }
+;; ```
+;;
+;;This dictionary describes two variables: `caseid` is a 12-character string that represents the respondent ID; `pregordr` is a one-byte integer that indicates which pregnancy this record
+;;describes for this respondent.
+;;
+;;The code you downloaded includes many functions used in this book, including functions that read the Stata dictionary
+;;and the NSFG data file. Here’s how they are used in `nsfg.clj`:
+;;
+;;```
+;; (defn read-fem-preg-dataset
+;;   "Read Stata data set, return an dataset."
+;;   [dict-path data-path]
+;;   (let [dict-path (or dict-path "resources/2002FemPreg.dct")
+;;         data-path (or data-path "resources/2002FemPreg.dat.gz")
+;;         dict   (read-dictionary dict-path)
+;;         header (map (comp keyword :name) dict)]
+;;     (with-open [r (reader data-path)]
+;;       (->> (tc/dataset (read-dictionary-data dict r)
+;;                        {:layout :as-rows
+;;                         :column-names header
+;;                         :dataset-name "2002FemPreg"})
+;;            (clean-fem-preg)))))
+;;```
+;;
+;;`read-dictionary` takes the name of the dictionary file and returns dict, a vector that contains the column information from the dictionary file.
+;;`read-dictionary-data` reads the data file using the dictionary.
+;;
+;;## 1.4 DataFrames
+;;
+;;The result of `tc/dataset` is a dataset (data frame), which is the columnar data structure provided by tablecloth, which is a Clojure data library we’ll use throughout this book.
+;;A DataFrame contains a row for each record, in this case one row per pregnancy, and a column for each variable.
+;;
+;;In addition to the data, a DataFrame also contains the variable names and their types, and it provides methods for accessing and modifying the data.
+;;If you print df you get a truncated view of the rows and columns, and the shape of the DataFrame, which is 13593 rows/records and 244 columns/variables.
+;;```
+;;user> (require '[data.nsfg :as nsfg])
```

Comment:
It is recommended to use Clay for input-output examples rather than including the REPL output yourself.

Reply:
Thanks for pointing that out, even better for me!